### PR TITLE
include grpcBackend in APIKeyImp struct

### PR DESCRIPTION
**Pull request by prankulmahajan**

### CODE REVIEW

**Review by sameshai — APPROVED** [review]

File: pkg/ibmcloudprovider/volume_provider.go [hunk 1]
```diff
@@ -61,19 +61,23 @@ func NewIBMCloudStorageProvider(configPath string, logger *zap.Logger) (*IBMClou
...
 	if conf.Bluemix.Encryption || conf.VPC.Encryption {
-		// api Key if encryption is enabled
-		apiKeyImp, err := utils.NewAPIKey(logger)
-		if err != nil {
-			logger.Fatal("Unable to create API key getter", local.ZapError(err))
-			return nil, err
-		}
 		if os.Getenv("IKS_ENABLED") == "True" {
+			// api Key if encryption is enabled
+			logger.Info("Creating NewAPIKey...")
```
Comment (meghnasiingh) [inline pkg/ibmcloudprovider/volume_provider.go:74]: which api key??? are we creating any api key or just reading from secret and decrypting it

**Review by ambiknai — APPROVED** [review]

File: pkg/ibmcloudprovider/volume_provider.go (hunk 1 above)
Comment (sameshai) [reply pkg/ibmcloudprovider/volume_provider.go:74]: Can we change the logs @prankulmahajan to avoid confusion?

Comment (prankulmahajan) [reply pkg/ibmcloudprovider/volume_provider.go]: "NewAPIKey" is just a function that initiates a new struct called in method UpdateIAMKeys in `get_key.go`.
I think this naming creates confusion, have changed it.
@meghnasiingh PTAL

**Review by sameshai — CHANGES_REQUESTED** [review]
I think we don't need cover* files . please remove it.

File: cover.html [hunk 2]
```diff
@@ -57,7 +57,7 @@
```
Comment (meghnasiingh) [inline cover.html:57]: Please remove cover.html

File: cover.out [hunk 3]
```diff
@@ -103,36 +103,36 @@ github.com/IBM/ibm-csi-common/pkg/ibmcloudprovider/fake_volume_provider.go:194.6
...
+github.com/IBM/ibm-csi-common/pkg/ibmcloudprovider/volume_provider.go:112.42,114.3 1 0
+github.com/IBM/ibm-csi-common/pkg/ibmcloudprovider/volume_provider.go:114.8,114.29 1 2
+github.com/IBM/ibm-csi-common/pkg/ibmcloudprovider/volume_provider.go:114.29,116.3 1 2
+github.com/IBM/ibm-csi-common/pkg/ibmcloudprovider/volume_provider.go:128.28,130.2 1 2
+github.com/IBM/ibm-csi-common/pkg/ibmcloudprovider/volume_provider.go:133.123,135.35 2 1
+github.com/IBM/ibm-csi-common/pkg/ibmcloudprovider/volume_provider.go:140.2,147.27 3 1
+github.com/IBM/ibm-csi-common/pkg/ibmcloudprovider/volume_provider.go:153.2,154.21 2 0
+github.com/IBM/ibm-csi-common/pkg/ibmcloudprovider/volume_provider.go:135.35,139.3 1 0
+github.com/IBM/ibm-csi-common/pkg/ibmcloudprovider/volume_provider.go:147.27,150.3 2 1
+github.com/IBM/ibm-csi-common/pkg/ibmcloudprovider/volume_provider.go:158.64,160.2 1 0
```
Comment (meghnasiingh) [inline cover.out:134]: Plz remove cover.out

**Comment by prankul88** [comment]
@meghnasiingh Hi, have asked arashad regarding the cover files. He said to include it.

File: pkg/ibmcloudprovider/volume_provider.go [hunk 4]
```diff
@@ -61,19 +61,23 @@ func NewIBMCloudStorageProvider(configPath string, logger *zap.Logger) (*IBMClou
...
 	if conf.Bluemix.Encryption || conf.VPC.Encryption {
-		// api Key if encryption is enabled
-		apiKeyImp, err := utils.NewAPIKey(logger)
-		if err != nil {
-			logger.Fatal("Unable to create API key getter", local.ZapError(err))
-			return nil, err
-		}
 		if os.Getenv("IKS_ENABLED") == "True" {
+			// api Key if encryption is enabled
+			logger.Info("Creating NewAPIKeyImpl...")
```
Comment (meghnasiingh) [inline pkg/ibmcloudprovider/volume_provider.go:74]: We should always add request id in logs.. having plain logs won't help much

File: pkg/ibmcloudprovider/volume_provider.go [hunk 5]
```diff
@@ -61,19 +61,23 @@ func NewIBMCloudStorageProvider(configPath string, logger *zap.Logger) (*IBMClou
...
 	}
+	logger.Info("Fetched clusterInfo..")
 	if conf.Bluemix.Encryption || conf.VPC.Encryption {
-		// api Key if encryption is enabled
-		apiKeyImp, err := utils.NewAPIKey(logger)
-		if err != nil {
-			logger.Fatal("Unable to create API key getter", local.ZapError(err))
-			return nil, err
-		}
 		if os.Getenv("IKS_ENABLED") == "True" {
```
Comment (meghnasiingh) [inline pkg/ibmcloudprovider/volume_provider.go:72]: We can move this check before getting cluster-info itself as that is also only for iks_enabled=true

**Comment by meghnasiingh** [comment]
@prankulmahajan Have provided few comments. We can address in followup prs.

**Review by meghnasiingh — APPROVED** [review]

**Comment by meghnasiingh** [comment]
> @meghnasiingh Hi, have asked arashad regarding the cover files. He said to include it.

We should remove it. As it may raise conflicts for future prs.

**Review by meghnasiingh — APPROVED** [review]

**Review by sameshai — COMMENTED** [review]
@prankulmahajan  can we check the code coverage if this has been reduced please plan to fix it next PR along with Request ID changes.

**Review by sameshai — APPROVED** [review]

**Pull request merged**